File: suffix_Array/LexicographicallyLargestSuffix.java
import java.util.Scanner;
import java.util.TreeSet;

public class LexicographicallyLargestSuffix {
    public static void main(String [] CSECU){
        System.out.println("Lets Code it.");

        Scanner Arnab = new Scanner(System.in);
        System.out.print("Enter the String: ");
        String mystr = Arnab.next();
        
        int length = mystr.length();
        System.out.println("The Lexicographically Smallest Suffix is : "+findLSS(mystr,length));
        Arnab.close();
    }
    public static String findLSS(String mystr,int length){
        TreeSet<String> myTree = new TreeSet<>();
        for(int a=0;a<length;a++){
            myTree.add(mystr.substring(a));
        }
        return myTree.last();
    }
}

File: suffix_Array/Kotlin/LexicographicallyLargestSuffix.kt
fun findLSS(mystr: String, length: Int): String {
    val myTree = sortedSetOf<String>()
    for (a in 0 until length) {
        myTree.add(mystr.substring(a))
    }
    return myTree.last()
}

fun main() {
    println("Lets Code it.")
    print("Enter the String: ")
    val mystr = readLine() ?: ""
    val length = mystr.length
    println("The Lexicographically Largest Suffix is : ${findLSS(mystr, length)}")
} 